File: Bridge/Resources/Uri.js
    Bridge.define("System.Uri", {
        ctor: function (uriString) {
            this.$initialize();
            this.absoluteUri = uriString;
        },

        $ctor1: function (baseUri, relativeUri) {
            this.$initialize();
            this.absoluteUri = baseUri.absoluteUri + relativeUri;
        },

        getAbsoluteUri: function () {
            return this.absoluteUri;
        },

        toString: function () {
            return this.absoluteUri;
        }
    });

File: Bridge/System/Uri.cs
using Bridge;

namespace System
{
    /// <summary>
    /// Provides an object representation of a uniform resource identifier (URI) and easy access to the parts of the URI.
    /// </summary>
    [External]
    [Reflectable]
    public class Uri
    {
        public extern Uri(string uriString);

        [Template("new System.Uri.$ctor1({baseUri}, {relativeUri})")]
        public extern Uri(Uri baseUri, string relativeUri);

        [Template("encodeURIComponent({str})")]
        public extern static string EscapeDataString(string str);

        public extern string AbsoluteUri
        {
            [Template("getAbsoluteUri()")]
            get;
        }
    }
}